### ISSUE TITLE
Bug when used with type families

### ISSUE THREAD

**Issue by oliverbunting** [kcsongor/generic-lens#45]
There is a bug when combined with type families:

```haskell

data IType (a :: Nat) = IType
  deriving Generic

data BadTest a = BadTest {name1 :: IType (2*a)}
  deriving Generic

data OKTest a = OKTest {name2 :: IType a}
  deriving Generic

fails = view (field @"name1")

ok = view (field @"name2")

```

> Couldn't match type ‘generic-lens-1.0.0.1:Data.Generics.Internal.Families.Changing.ReplaceArgs
  (BadTest a)
                   (generic-lens-1.0.0.1:Data.Generics.Internal.Families.Changing.Unify
                                (2
                                 * generic-lens-1.0.0.1:Data.Generics.Internal.Families.Changing.P
                                     0
                                     a
                                     'generic-lens-1.0.0.1:Data.Generics.Internal.Families.Changing.PTag)
                                (2 * a)
                              generic-lens-1.0.0.1:Data.Generics.Internal.Families.Changing.++ '[])’
                     with ‘BadTest a’
        arising from a use of ‘field’

**Comment by kcsongor** [comment]
Indeed, your example does not work, as explained in this comment: https://github.com/kcsongor/generic-lens/issues/40#issuecomment-376290272

TL;DR: it _should_ work if you use the monomorphic `field'` lens, as that one doesn't try to be clever and find out if the type parameter can be changed.

If it doesn't work with `field'`, then that's a bug. Could you please try with that?
On that note, I think this should go somewhere in the documentation...

Thanks!

**Comment by oliverbunting** [comment]
That does indeed make the problem go away. Thank you!
